test/integration/datatypes_test.exs

```diff
@@ -192,4 +192,48 @@ defmodule DataTypesTest do
     assert Map.fetch!(row, "map_of_int_to_text") == nil
     assert Map.fetch!(row, "set_of_int") == nil
   end
+
+  test "user-defined types", %{conn: conn} do
+    statement = """
+    CREATE TYPE IF NOT EXISTS full_name
+    (first_name text,
+     last_name text)
+    """
+    Xandra.execute!(conn, statement)
+
+    statement = """
+    CREATE TYPE IF NOT EXISTS profile
+    (nickname text,
+     full_name frozen<full_name>)
+    """
+    Xandra.execute!(conn, statement)
+
+    statement = """
+    CREATE TABLE users
+    (id int PRIMARY KEY,
+     profile frozen<profile>)
+    """
+    Xandra.execute!(conn, statement)
+
+    statement = "INSERT INTO users (id, profile) VALUES (?, ?)"
+    foo_profile = %{
+      "nickname" => "foo",
+      "full_name" => %{"first_name" => "Kung", "last_name" => "Foo"},
+    }
+    bar_profile = %{
+      "nickname" => "bar",
```

Comment:
I still think `username` reads nicer, but feel free to merge, not an important detail :)